labman/db/tests/integration_tests.py

```diff
@@ -0,0 +1,213 @@
+        input_compositions, {"function": "amplicon_pool", "parameters": {}})
+    return pool_process
+
+
+def create_sequencing_process(user, pools):
+    seq_process = SequencingProcess.create(
+        user, pools, 'New sequencing run %s' % datetime.now(),
+        'Run experiment %s' % datetime.now(), Equipment(18), 151, 151,
+        User('admin@foo.bar'),
+        contacts=[User('test@foo.bar'), User('demo@microbio.me')])
+    return seq_process
+
+
+def amplicon_workflow(user, samples):
+    # Sample Plating
+    sp_process, sample_plate = create_sample_plate_process(user, samples[:96])
+    # gDNA extraction
+    ext_process, gdna_plate = create_gdna_extraction_process(
+        user, sample_plate)
+    # Amplicon library prep
+    amplicon_process, amplicon_plate = create_amplicon_prep(user, gdna_plate)
+    # Library plate quantification
+    amplicon_quant_process = create_quantification_process(
+        user, amplicon_plate)
+    # Plate pooling process
+    plate_pool_process = create_plate_pool_process(
+        user, amplicon_quant_process, amplicon_plate,
+        {'function': 'amplicon',
+         'parameters': {"dna_amount": 240, "min_val": 1, "max_val": 15,
+                        "blank_volume": 2, "robot": 6, "destination": 1}})
+    # Quantify pools
+    pool_quant_process = create_pool_quantification_process(
+        user, [plate_pool_process.pool])
+    # Create sequencing pool process
+    seq_pool_process = create_pools_pool_process(
+        user, pool_quant_process, [plate_pool_process.pool])
+    # Sequencing process
+    create_sequencing_process(user, [seq_pool_process.pool])
+
+
+def shotgun_workflow(user, samples):
+    # Sample Plating
+    sp_process, sample_plate = create_sample_plate_process(user, samples[:96])
+    # gDNA extraction
+    ext_process, gdna_plate = create_gdna_extraction_process(
+        user, sample_plate)
+    # gDNA compression
+    comp_process, compressed_plate = create_compression_process(
+        user, [gdna_plate])
+    # gDNA compressed quantification
+    gdna_comp_quant_process = create_quantification_process(
+        user, compressed_plate)
+    # Normalization process
+    norm_process, norm_plate = create_normalization_process(
+        user, gdna_comp_quant_process)
+    # Library prep shotgun
+    shotgun_process, shotgun_plate = create_shotgun_process(user, norm_plate)
+    # Quantify library plate
+    shotgun_quant_process = create_quantification_process(user, shotgun_plate)
+    # Pooling process
+    pool_process = create_plate_pool_process(
+        user, shotgun_quant_process, shotgun_plate,
+        {'function': 'equal', 'parameters': {'total_vol': 60, 'size': 500}})
+    # Sequencing process
+    create_sequencing_process(user, [pool_process.pool])
+
+
+@click.command()
+def integration_tests():
+    samples = get_samples()
+    user = User('test@foo.bar')
+    amplicon_workflow(user, samples)
+    shotgun_workflow(user, samples)
+
```

Comment:
would it be possible to assert the sample sheets contain the expected values?

Reply:
Done - I had to modify the code to ensure the sorting of the samples in the sample sheet.